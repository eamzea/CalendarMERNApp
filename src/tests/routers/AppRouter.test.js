import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import AppRouter from "../../routers/AppRouter";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe("Pruebas en AppRouter", () => {
  test("Debe de mostrar el Loading", () => {
    const initState = {
      auth: {
        checking: true,
      },
    };
    let store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar Public Route", () => {
    const initState = {
      auth: {
        checking: false,
        uid: null,
      },
    };
    let store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".login-container").exists()).toBe(true);
  });

  test("Debe mostrar Private Route", () => {
    const initState = {
      ui: {
        modalOpen: false,
      },
      calendar: {
        events: [],
      },
      auth: {
        checking: false,
        uid: "123",
        name: "Edgar",
      },
    };
    let store = mockStore(initState);

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".calendar-screen").exists()).toBe(true);
  });
});
