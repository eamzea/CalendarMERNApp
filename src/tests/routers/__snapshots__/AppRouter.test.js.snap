// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter Debe de mostrar el Loading 1`] = `
<div
  className="load__container"
>
  <div
    className="load__container-text"
  >
    <p
      className="load__text"
    >
      Loading...
    </p>
  </div>
  <div
    className="load__container-spinner"
  >
    <i
      className="fas fa-circle-notch fa-spin fa-10x load__spinner"
    />
  </div>
</div>
`;

exports[`Pruebas en AppRouter Debe mostrar Private Route 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        Hello 
        Edgar
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        <span>
           Logout
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          October 2020
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        className="rbc-month-view"
      >
        <div
          className="rbc-row rbc-month-header"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span>
              Sun
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span>
              Mon
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span>
              Tue
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span>
              Wed
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span>
              Thu
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span>
              Fri
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span>
              Sat
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  03
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  07
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  10
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  14
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  17
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  21
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  24
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  28
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  31
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
  </div>
</div>
`;

exports[`Pruebas en AppRouter Debe mostrar Public Route 1`] = `
<div>
  <div
    className="container login-container d-flex justify-content-center align-items-center rounded"
  >
    <div
      className="row divs-container"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Login
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lEmail"
              onChange={[Function]}
              placeholder="your@email.com"
              type="text"
              value="email@email.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lPassword"
              onChange={[Function]}
              placeholder="*******"
              type="password"
              value="123456"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rName"
              onChange={[Function]}
              placeholder="Edgar"
              type="text"
              value="Arturo"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rEmail"
              onChange={[Function]}
              placeholder="tu@correo.com"
              type="email"
              value="arturo@arturo.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="123456"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword2"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value="123456"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Register"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
